### ISSUE TITLE
exception thrown from xmlhttprequest

### ISSUE THREAD

**Issue by davidbielik** [tmpvar/jsdom#1808]
I'm getting some exceptions from various websites, box.com is one example. The code below throws an error which originates from the `xmlhttprequest` module. 

Can someone suggest how to fix, prevent, or catch this so it doesn't break the rest of my script?

Thanks!!

```
jsdom.env({ 
    url: 'http://box.com', 
    done: function(error, window){  
        console.log(error || window.document.title); 
    }, 
    strictSSL: false, 
    features: {
        FetchExternalResources: ['script'],
        ProcessExternalResources: ['script'],
        SkipExternalResources: false
    }
});
```

```
TypeError: Cannot read property 'on' of null
    at receiveResponse (/test/node_modules/jsdom/lib/jsdom/living/xmlhttprequest.js:921:22)
    at Request.client.on.res (/test/node_modules/jsdom/lib/jsdom/living/xmlhttprequest.js:691:38)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:191:7)
    at Request.onRequestResponse (/test/node_modules/request/request.js:1074:10)
    at emitOne (events.js:96:13)
    at ClientRequest.emit (events.js:191:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:522:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
    at TLSSocket.socketOnData (_http_client.js:411:20)
```

**Comment by domenic** [comment]
I've been debugging this a bit. It appears the problem is that when the XHR gets abort()ed, it sets the properties.client to null. Then receiveResponse tries to use the client, but it doesn't exist.

I tried bailing out early in receiveResponse, but then the program just hung forever. Still investigating...

**Comment by domenic** [comment]
This will be fixed in v10.1.0. The hanging forever was because box.com has a setInterval that runs forever, so that's expected; you need to use window.close() to shut things down. A test case that works well on master right now:

```js
"use strict";
const jsdom = require("jsdom/lib/old-api.js");

jsdom.env({
    url: 'http://box.com',
    done: function(error, window){
        console.log(error || window.document.title);
        window.close();
    },
    strictSSL: false,
    features: {
        FetchExternalResources: ['script'],
        ProcessExternalResources: ['script'],
        SkipExternalResources: false
    }
});
```
